src/Microsoft.NetCore.Analyzers/Microsoft.NetCore.Analyzers.md

```diff
@@ -33,33 +33,46 @@ Sr. No. | Rule ID | Title | Category | Enabled | CodeFix | Description |
+58 | CA5358 | Do Not Use Unsafe Cipher Modes | Security | False | False | These modes are vulnerable to attacks. Use only approved modes (CBC, CTS). |
+59 | CA5359 | Do Not Disable Certificate Validation | Security | True | False | A certificate can help authenticate the identity of the server. Clients should validate the server certificate to ensure requests are sent to the intended server. If the ServerCertificateValidationCallback always returns 'true', any certificate will pass validation. |
+60 | CA5360 | Do Not Call Dangerous Methods In Deserialization | Security | True | False | Insecure Deserialization is a vulnerability which occurs when untrusted data is used to abuse the logic of an application, inflict a Denial-of-Service (DoS) attack, or even execute arbitrary code upon it being deserialized. It’s frequently possible for malicious users to abuse these deserialization features when the application is deserializing untrusted data which is under their control. Specifically, invoke dangerous methods in the process of deserialization. Successful insecure deserialization attacks could allow an attacker to carry out attacks such as DoS attacks, authentication bypasses, and remote code execution. |
+61 | CA5361 | Do Not Disable SChannel Use of Strong Crypto | Security | True | False | Starting with the .NET Framework 4.6, the System.Net.ServicePointManager and System.Net.Security.SslStream classes are recommeded to use new protocols. The old ones have protocol weaknesses and are not supported. Setting Switch.System.Net.DontEnableSchUseStrongCrypto with true will use the old weak crypto check and opt out of the protocol migration. |
+62 | CA5362 | Do Not Refer Self In Serializable Class | Security | False | False | This can allow an attacker to DOS or exhaust the memory of the process. |
+63 | CA5363 | Do Not Disable Request Validation | Security | True | False | Request validation is a feature in ASP.NET that examines HTTP requests and determines whether they contain potentially dangerous content. This check adds protection from markup or code in the URL query string, cookies, or posted form values that might have been added for malicious purposes. So, it is generally desirable and should be left enabled for defense in depth. |
+64 | CA5364 | Do Not Use Deprecated Security Protocols | Security | True | False | Using a deprecated security protocol rather than the system default is risky. |
+65 | CA5365 | Do Not Disable HTTP Header Checking | Security | True | False | HTTP header checking enables encoding of the carriage return and newline characters, \r and \n, that are found in response headers. This encoding can help to avoid injection attacks that exploit an application that echoes untrusted data contained by the header. |
+66 | CA5366 |  | Security | True | False |  |
```

Comment:
>   [](start = 13, length = 2)

This is weird.

Reply:
Yes, same thing with few more rules below. Wondering why the title couldn't be accessed off the descriptor: https://github.com/dotnet/roslyn-analyzers/blob/master/src/GenerateAnalyzerRulesets/Program.cs#L395

Reply:
Seems like the case only for analyzers deriving from `UseXmlReaderBase`

Reply:
I think I figured it out. [GetResourceString](https://github.com/dotnet/roslyn-analyzers/blob/020a095feae4881ea5a13f8ed40c56e478f6a67a/src/Microsoft.NetCore.Analyzers/Core/Security/Helpers/SecurityHelpers.cs#L79) helper method is only expecting resource strings from `MicrosoftNetCoreSecurityResources`, but rules deriving from `UseXmlReaderBase` are passing in resource strings from `SystemSecurityCryptographyResources`. @dotpaul, you want me to file a bug for you to fix this?

Reply:
Also, @dotpaul if you don't mind, I would still like to go ahead with this PR and you can fix this as a follow up. Just run CIBuild locally to regenerate this file after your fix.

Reply:
File an issue for @LingxiaChen, please

Reply:
https://github.com/dotnet/roslyn-analyzers/issues/2421